engine/tests/vm/test_redirect_precedence.py

```
@@ -0,0 +1,382 @@
+
+
+# ---------------------------------------------------------------------------
+# The ladder
+# ---------------------------------------------------------------------------
+
+
+class TestRedirectLadder:
+    """Which layer's claim survives when several handlers all return an edge."""
+
+    def test_global_beats_every_later_layer(self, graph, crossing, null_ctx):
+        origin, regular, collapse, outro = crossing
```

Ruff (0.16.1)

[warning] Unpacked variable `regular` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)